xds/internal/clients/xdsclient/ads_stream.go

```diff
@@ -444,6 +445,9 @@ func (s *adsStreamImpl) sendMessageLocked(stream clients.Stream, names []string,
 		}
 	}
 
+	// Call the event handler to remove unsubscribed cache entries.
+	s.eventHandler.onRequiredToRemoveUnsubscribedCacheEntries(url)
```

Comment:
Should this happen *only* if the call to `Send` below is successful?

Reply:
> we want to ensure the cache entries are deleted even if discovery request fails. In case of failure when the stream restarts, nonce is reset anyways.

I think this is good information that is worth capturing in a comment. Thanks.